src/test/java/com/scottlogic/filters/DatesFilterTest.java

```diff
@@ -0,0 +1,116 @@
+        List<UserPost> userPosts = Arrays.asList(userPost1, userPost2, userPost3);
+        List<UserPost> actualResult = filterByDates.filter(userPosts);
+        List<UserPost> expectedResult = Arrays.asList(userPost1, userPost2, userPost3);
+        Assert.assertEquals(actualResult, expectedResult);
+    }
+
+    @Test
+    public void filter_extremeDatesInInput_filteredListReturned() {
+        UserPost userPost1 = createUserPost(OffsetDateTime.MIN);
+        UserPost userPost2 = createUserPost(OffsetDateTime.MAX);
+        UserPost userPost3 = createUserPost(OffsetDateTime.of(2020, 1, 3, 7, 12, 3, 1, ZoneOffset.UTC));
+        UserPost userPost4 = createUserPost(OffsetDateTime.of(2012, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC));
+
+        DatesFilter filterByDates = new DatesFilter(OffsetDateTime.of(1800, 1, 1, 7, 12, 3, 0, ZoneOffset.UTC), OffsetDateTime.of(2800, 12, 3, 7, 12, 3, 0, ZoneOffset.UTC));
+        List<UserPost> userPosts = Arrays.asList(userPost1, userPost2, userPost3, userPost4);
+        List<UserPost> actualResult = filterByDates.filter(userPosts);
+        List<UserPost> expectedResult = Arrays.asList(userPost3, userPost4);
+        Assert.assertEquals(actualResult, expectedResult);
+    }
+
+    @Test
+    public void filter_borderDatesMax_borderDatesIncludedReturned() {
+        UserPost userPost1 = createUserPost(OffsetDateTime.of(20, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC));
+        UserPost userPost2 = createUserPost(OffsetDateTime.of(2020, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC));
+        UserPost userPost3 = createUserPost(OffsetDateTime.of(2020, 1, 3, 7, 12, 3, 1, ZoneOffset.UTC));
+        UserPost userPost4 = createUserPost(OffsetDateTime.of(2012, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC));
+        UserPost userPost5 = createUserPost(OffsetDateTime.of(2020, 12, 31, 7, 12, 3, 0, ZoneOffset.UTC));
+
+        DatesFilter filterByDates = new DatesFilter(OffsetDateTime.of(2020, 1, 1, 7, 12, 3, 0, ZoneOffset.UTC), OffsetDateTime.of(2020, 12, 31, 7, 12, 3, 0, ZoneOffset.UTC));
+        List<UserPost> userPosts = Arrays.asList(userPost1, userPost2, userPost3, userPost4, userPost5);
+        List<UserPost> actualResult = filterByDates.filter(userPosts);
+        List<UserPost> expectedResult = Arrays.asList(userPost2, userPost3, userPost5);
+        Assert.assertEquals(actualResult, expectedResult);
+    }
+
+    @Test
+    public void filter_borderDatesMin_borderDatesIncludedReturnedSecondsIgnored() {
+        UserPost userPost1 = createUserPost(OffsetDateTime.of(20, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC));
+        UserPost userPost2 = createUserPost(OffsetDateTime.of(2020, 1, 3, 7, 12, 4, 0, ZoneOffset.UTC));
+        UserPost userPost3 = createUserPost(OffsetDateTime.of(2020, 1, 3, 7, 12, 3, 1, ZoneOffset.UTC));
+        UserPost userPost4 = createUserPost(OffsetDateTime.of(2012, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC));
```

Comment:
It would be a lot easier to see what's going on if we named these posts a bit more descriptively. It would also be nice to extract the repeated date into a variable, then use it for setting up both the post and the filter.